### PR TITLE
Feature/remote controlling

### PR DESCRIPTION
**Pull request by Xely**

### CODE REVIEW

**Review by karakakou0 — DISMISSED** [review]

**Comment by foch01** [comment]
Github action & codacy failed

File: client/src/core/services/robiot.service.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        if (config.id == 301) {
+            return config?.content.split(',')[1];
+        }
+    }
+
+    getConfig(id: number): Observable<Configuration> {
+        return this.http.get<Configuration>(this.resourceUrl + '/robiot/' + id);
+    }
+
+    saveConfig(config: Configuration) {
```
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:40]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/map-display/map-display.component.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    selector: 'app-map-display',
+    templateUrl: './map-display.component.html',
+    styleUrls: ['./map-display.component.scss'],
+})
+export class MapDisplayComponent implements OnInit {
+    @Input() events: Observable<void>;
+    private updateMap: Subscription;
+
+    public map: Array<Array<string>>;
+    public rows: Array<any>;
```
Comment (MaolmeoX) [inline client/src/app/map-display/map-display.component.ts:19]: Codacy found an issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/map-display/map-display.component.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        this.robiotService = robiotService;
+    }
+
+    ngOnInit(): void {
+        this.createMap();
+
+        this.updateMap = this.events?.subscribe(() => this.createMap());
+    }
+
+    createMap() {
```
Comment (MaolmeoX) [inline client/src/app/map-display/map-display.component.ts:34]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: api/src/controllers/robiot.controller.ts [hunk 4]
```diff
@@ -1,30 +1,44 @@
-import { Request, Response, NextFunction } from 'express';
+import { NextFunction, Request, Response } from 'express';
```
Comment (MaolmeoX) [inline api/src/controllers/robiot.controller.ts:1]: Codacy found an issue: ['NextFunction' is defined but never used.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        for (const tree of terrain.trees) {
+            map[tree.x][tree.y] = CaseValue.TREE;
+        }
+
+        return map;
+    }
+
+    // Checks if a destination is obtainable (not out of bound, doesnt cross obstacles etc..)
+    checkDestinationValidity(
+        map,
```
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:93]: Codacy found an issue: [Argument 'map' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/remote-controlling/remote-controlling.component.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    // Get the data necessary for our component
+    initData() {
+        this.getConfigData();
+        this.terrain = terrain1;
+        this.map = this.robiotService.buildArrayFromTerrain(this.terrain);
+    }
+
+    getConfigData() {
```
Comment (MaolmeoX) [inline client/src/app/remote-controlling/remote-controlling.component.ts:89]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        coordToChange,
```
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:94]: Codacy found an issue: [Argument 'coordToChange' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/dijkstras.service.ts [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+})
+export class DijkstrasService {
+    useDijkstrasAlgorithm(
+        map: Array<Array<string>>,
+        position: Position,
+        treesToCheck: Position[],
+    ): { map: Array<Array<{ distance: number; terrain: string }>>; report: string } {
+        let trees = treesToCheck;
+        let newMap;
+        let report: string = '';
```
Comment (MaolmeoX) [inline client/src/core/services/dijkstras.service.ts:16]: Codacy found an issue: [Type string trivially inferred from a string literal, remove type annotation.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/remote-controlling/remote-controlling.component.scss [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+.status-title {
+    font-weight: bold;
+    font-size: 16px;
+    margin-right: 20px;
+    margin-left: 30px;
+}
+
+app-map-display {
```
Comment (MaolmeoX) [inline client/src/app/remote-controlling/remote-controlling.component.scss:32]: Codacy found an issue: [Unexpected unknown type selector "app-map-display" (selector-type-no-unknown)](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+        oldValue,
```
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:95]: Codacy found an issue: [Argument 'oldValue' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/remote-controlling/remote-controlling.component.ts [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+                this.displayInfoMessage(data['message']['content']);
+            });
+        } else {
+            // If the destination isn't valid, display a message
+            this.displayInfoMessage(validDestination.msg);
+        }
+    }
+
+    // Get the data necessary for our component
+    initData() {
```
Comment (MaolmeoX) [inline client/src/app/remote-controlling/remote-controlling.component.ts:83]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: api/src/controllers/robiot.controller.ts [hunk 12]
```diff
@@ -1,30 +1,44 @@
...
 import '../config/passport';
+import { getRobiotBattery, getRobiotPosition, getRobiotStatus, moveRobiot } from '../services/robiot.service';
 import logger from '../util/logger';
 
 /**
  * GET /api/v1/robiot/
  * get Config list from one ROBIOT
  */
-export const getConfigListFromRobiot = async (req: Request, res: Response, next: NextFunction) => {
+export const getConfigListFromRobiot = async (req: Request, res: Response) => {
```
Comment (MaolmeoX) [inline api/src/controllers/robiot.controller.ts:10]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/dijkstras/dijkstras.component.ts [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+import { RobiotService } from '../../core/services/robiot.service';
+
+@Component({
+    selector: 'app-dijkstras',
+    templateUrl: './dijkstras.component.html',
+    styleUrls: ['./dijkstras.component.scss'],
+})
+export class DijkstrasComponent implements OnInit {
+    public map: Array<Array<string>>;
+    public result: { map: Array<Array<any>>; report: string };
```
Comment (MaolmeoX) [inline client/src/app/dijkstras/dijkstras.component.ts:13]: Codacy found an issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/map-display/map-display.component.scss [hunk 14]
```diff
@@ -0,0 +1,41 @@
+.title, .legend {
```
Comment (MaolmeoX) [inline client/src/app/map-display/map-display.component.scss:1]: Codacy found an issue: [Expected newline after "," (selector-list-comma-newline-after)](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/map-display/map-display.component.ts [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+    templateUrl: './map-display.component.html',
+    styleUrls: ['./map-display.component.scss'],
+})
+export class MapDisplayComponent implements OnInit {
+    @Input() events: Observable<void>;
+    private updateMap: Subscription;
+
+    public map: Array<Array<string>>;
+    public rows: Array<any>;
+    public columns: Array<any>;
```
Comment (MaolmeoX) [inline client/src/app/map-display/map-display.component.ts:20]: Codacy found an issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/dijkstras.service.ts [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+            trees = trees.filter((tree) => tree.x != closestTree.x || tree.y != closestTree.y);
+        }
+
+        report += 'No more tree to check, going back to base.';
+
+        return { map: newMap, report: report };
+    }
+
+    // Fills the map object with value indicating the distance from the initial position
+    fillEveryDirection(map: Array<Array<any>>, position: Position, turn: number) {
```
Comment (MaolmeoX) [inline client/src/core/services/dijkstras.service.ts:44]: Codacy found an issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+        newValue,
```
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:96]: Codacy found an issue: [Argument 'newValue' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+        otherCoordValue,
```
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:97]: Codacy found an issue: [Argument 'otherCoordValue' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/dijkstras.service.ts (hunk 16 above)
Comment (MaolmeoX) [inline client/src/core/services/dijkstras.service.ts:44]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/remote-controlling/remote-controlling.component.ts [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+            }, 1000);
+        });
+    }
+
+    canSubmit(): boolean {
+        // If form is valid and robot isn't stopped
+        return this.form.valid && this.statusConfig.content == 'stopped';
+    }
+
+    displayInfoMessage(msg: string) {
```
Comment (MaolmeoX) [inline client/src/app/remote-controlling/remote-controlling.component.ts:114]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/remote-controlling/remote-controlling.component.ts [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+    ngOnInit(): void {
+        this.initData();
+
+        this.form = this.formBuilder.group({
+            coord: new FormControl('x', Validators.required),
+            value: new FormControl('', [Validators.pattern(this.coordValueRegex), Validators.required]),
+        });
+    }
+
+    submitForm() {
```
Comment (MaolmeoX) [inline client/src/app/remote-controlling/remote-controlling.component.ts:41]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts (hunk 17 above)
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:96]: Codacy found an issue: [Argument 'newValue' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/map-display/map-display.component.ts (hunk 2 above)
Comment (MaolmeoX) [inline client/src/app/map-display/map-display.component.ts:19]: Codacy found an issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/remote-controlling/remote-controlling.component.ts (hunk 19 above)
Comment (MaolmeoX) [inline client/src/app/remote-controlling/remote-controlling.component.ts:114]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: api/src/controllers/robiot.controller.ts (hunk 12 above)
Comment (MaolmeoX) [inline api/src/controllers/robiot.controller.ts:10]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: api/src/controllers/robiot.controller.ts (hunk 4 above)
Comment (MaolmeoX) [inline api/src/controllers/robiot.controller.ts:1]: Codacy found an issue: ['NextFunction' is defined but never used.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/dijkstras.service.ts (hunk 16 above)
Comment (MaolmeoX) [inline client/src/core/services/dijkstras.service.ts:44]: Codacy found an issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/remote-controlling/remote-controlling.component.ts (hunk 6 above)
Comment (MaolmeoX) [inline client/src/app/remote-controlling/remote-controlling.component.ts:89]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/dijkstras.service.ts (hunk 8 above)
Comment (MaolmeoX) [inline client/src/core/services/dijkstras.service.ts:16]: Codacy found an issue: [Type string trivially inferred from a string literal, remove type annotation.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts (hunk 18 above)
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:97]: Codacy found an issue: [Argument 'otherCoordValue' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/remote-controlling/remote-controlling.component.ts (hunk 20 above)
Comment (MaolmeoX) [inline client/src/app/remote-controlling/remote-controlling.component.ts:41]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/remote-controlling/remote-controlling.component.ts (hunk 11 above)
Comment (MaolmeoX) [inline client/src/app/remote-controlling/remote-controlling.component.ts:83]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts (hunk 5 above)
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:93]: Codacy found an issue: [Argument 'map' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts (hunk 1 above)
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:40]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/map-display/map-display.component.ts (hunk 3 above)
Comment (MaolmeoX) [inline client/src/app/map-display/map-display.component.ts:34]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts (hunk 10 above)
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:95]: Codacy found an issue: [Argument 'oldValue' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/map-display/map-display.component.ts (hunk 15 above)
Comment (MaolmeoX) [inline client/src/app/map-display/map-display.component.ts:20]: Codacy found an issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/dijkstras.service.ts (hunk 16 above)
Comment (MaolmeoX) [inline client/src/core/services/dijkstras.service.ts:44]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/core/services/robiot.service.ts (hunk 7 above)
Comment (MaolmeoX) [inline client/src/core/services/robiot.service.ts:94]: Codacy found an issue: [Argument 'coordToChange' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

File: client/src/app/dijkstras/dijkstras.component.ts (hunk 13 above)
Comment (MaolmeoX) [inline client/src/app/dijkstras/dijkstras.component.ts:13]: Codacy found an issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

Comment (MaolmeoX) [inline client/src/app/dijkstras/dijkstras.component.ts:13]: Codacy found an issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)

**Review by MaolmeoX — DISMISSED** [review]

**Review by karakakou0 — DISMISSED** [review]

**Review by MaolmeoX — DISMISSED** [review]

File: api/package.json [hunk 21]
```diff
@@ -7,7 +7,7 @@
   "license": "MIT",
   "scripts": {
     "start": "npm run build && npm run serve",
-    "build": "npm run build-ts && npm run lint",
+    "build": "npm run build-ts",
```
Comment (foch01) [inline api/package.json]: Garder npm run lint

**Review by foch01 — CHANGES_REQUESTED** [review]

**Review by foch01 — APPROVED** [review]

**Review by MaolmeoX — APPROVED** [review]

**Pull request merged**